lib/elixir/lib/kernel.ex

```diff
@@ -217,6 +217,8 @@ defmodule Kernel do
   # We need this check only for bootstrap purposes.
   # Once Kernel is loaded and we recompile, it is a no-op.
   @compile {:inline, bootstrapped?: 1}
+  @dialyzer {:nowarn_function, bootstrapped?: 1}
+  @spec bootstrapped?(module) :: boolean
```

Comment:
The fact that `bootstrapped?` is getting inlined is causing multiple `match` warnings in this file. These changes didn't help. I tried ignoring the warning at each call site but I couldn't figure out how to do that for `defmacro @{a, b, c}`. I'll keep investigating.